File: app/src/main/java/com/ed/redditapp/ui/imageview/ImageViewDaggerModule.java
package com.ed.redditapp.ui.imageview;

import com.ed.redditapp.ActivityScope;

import dagger.Module;
import dagger.Provides;

@Module
public class ImageViewDaggerModule {
    private final ImageViewActivity activity;

    public ImageViewDaggerModule(ImageViewActivity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityScope
    public ImageViewModel provideModel() {
        return new ImageViewModel();
    }

    @Provides
    @ActivityScope
    public ImageViewPresenter providePresenter(ImageViewModel model) {
        return new ImageViewPresenter(activity, model);
    }
}

File: app/src/main/java/com/ed/redditapp/ui/imageview/ImageViewDaggerModule.kt
package com.ed.redditapp.ui.imageview

import com.ed.redditapp.ActivityScope
import dagger.Module
import dagger.Provides

@Module
class ImageViewDaggerModule(private val activity: ImageViewActivity) {
    @Provides
    @ActivityScope
    fun model() = ImageViewModel()

    @Provides
    @ActivityScope
    fun presenter(model: ImageViewModel) = ImageViewPresenter(activity, model)
}